libs/clients/middlewares/src/lib/withMetrics.ts

```diff
@@ -0,0 +1,40 @@
+import { dogStatsD } from '@island.is/infra-metrics'
+
+import { FetchAPI } from './nodeFetch'
+
+export interface MetricsMiddlewareOptions {
+  fetch: FetchAPI
+  name: string
+}
+
+export const withMetrics = ({
+  fetch,
+  name,
+}: MetricsMiddlewareOptions): FetchAPI => {
```

Comment:
Sorry, one more thing for your "simple example PR". :D 

It would be reaaaally good to add tests for this. At least that 500 responses trigger the correct increments. Depending on the ordering of the middlewares, 500 responses are returned normally or thrown as an error.

I think it's not an issue as currently ordered, but this could silently break if someone reorders middlewares in the future, unless we have a test for it.

Reply:
I added tests, can you review again?